backend_new/src/services/unit-rent-type.service.ts

```diff
@@ -0,0 +1,112 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from './prisma.service';
+import { UnitRentType } from '../dtos/unit-rent-types/unit-rent-type.dto';
+import { UnitRentTypeCreate } from '../dtos/unit-rent-types/unit-rent-type-create.dto';
+import { UnitRentTypeUpdate } from '../dtos/unit-rent-types/unit-rent-type-update.dto';
+import { mapTo } from '../utilities/mapTo';
+import { SuccessDTO } from '../dtos/shared/success.dto';
+
+/*
+  this is the service for unit rent types
+  it handles all the backend's business logic for reading/writing/deleting unit rent type data
+*/
+
+@Injectable()
+export class UnitRentTypeService {
+  constructor(private prisma: PrismaService) {}
+
+  /*
+    this will get a set of unit rent types given the params passed in
+  */
+  async list(): Promise<UnitRentType[]> {
+    const rawUnitRentTypes = await this.prisma.unitRentTypes.findMany();
+    return mapTo(UnitRentType, rawUnitRentTypes);
+  }
+
+  /*
+    this will return 1 unit rent type or error
+  */
+  async findOne(unitRentTypeId: string): Promise<UnitRentType> {
+    const rawUnitRentType = await this.prisma.unitRentTypes.findFirst({
+      where: {
+        id: {
+          equals: unitRentTypeId,
+        },
+      },
+    });
```

Comment:
Same here about the `findUnique`

Reply:
Actually, this findOne function is almost identical to the `findOrThrow` function below. Could you change the return of that function to the raw unit type and then this function calls that followed by the `mapTo` function?

Reply:
updated!